app/controllers/chargeback_controller.rb

```diff
@@ -208,28 +228,48 @@ def cb_rate_edit
       if params[:typ] == "copy" # if tab was not changed
         session[:changed] = true
         @sb[:rate_details] = []
+        @sb[:num_tiers] = []
+        @sb[:tiers] = []
         rate = ChargebackRate.find(obj[0])
         @sb[:rate] = ChargebackRate.new
         @sb[:rate].description = "Copy of " + rate.description
         @sb[:rate].rate_type = rate.rate_type
-        rate_details = rate.chargeback_rate_details
+        rate_details = rate.chargeback_rate_details.to_a
+        rate_details.sort_by! { |rd| [rd.group.downcase, rd.description.downcase] }
         # Create new rate detail records for copied rate record
-        rate_details.each do |r|
+        rate_details.each_with_index do |r, i|
+          @sb[:tiers][i] = []
           detail = ChargebackRateDetail.new
           detail.description = r[:description]
           detail.source = r[:source]
-          detail.rate = r[:rate]
           detail.per_time = r[:per_time]
           detail.group = r[:group]
           detail.per_unit = r[:per_unit]
           detail.metric = r[:metric]
           detail.chargeback_rate_detail_measure_id = r[:chargeback_rate_detail_measure_id]
+          r.chargeback_tiers.each do |t|
+            tier = ChargebackTier.new
+            tier.start = t[:start]
+            tier.end = t[:end]
+            tier.fix_rate = t[:fix_rate]
+            tier.var_rate = t[:var_rate]
+            tier.chargeback_rate_detail_id = detail.id
+            @sb[:tiers][i].push(tier)
+          end
           @sb[:rate_details].push(detail) unless @sb[:rate_details].include?(detail)
+          @sb[:num_tiers][i] = @sb[:tiers][i].length
         end
       else
         session[:changed] = false
         @sb[:rate] = params[:typ] == "new" ? ChargebackRate.new : ChargebackRate.find(obj[0])
+        @sb[:num_tiers] = []
+        @sb[:tiers] = []
         @sb[:rate_details] = @sb[:rate].chargeback_rate_details.to_a
+        @sb[:rate_details].sort_by! { |rd| [rd[:group].downcase, rd[:description].downcase] }
+        @sb[:rate_details].each_with_index do |detail, i|
+          @sb[:tiers][i] = detail.chargeback_tiers.to_a
+          @sb[:num_tiers][i] = detail.chargeback_tiers.to_a.length
```

Comment:
Please use `#size`, not `#count` or `#length`, as that method differs to ActiveRecord as to whether or not a query/load is needed to calculate (the others force a separate query and/or load). Even if it doesn't make a difference in this case, good to have that in the off chance it does one day.